app/server/contactUsApi.ts

```diff
@@ -50,18 +51,39 @@ export const contactUsFormHandler = async (req: Request, res: Response) => {
     });
 };
 
-const parseAndValidate = (body: any): ContactUsReq | undefined => {
+const parseAndValidate = async (
+  body: any
+): Promise<ContactUsReq | undefined> => {
   try {
     const bodyAsJson = body ? JSON.parse(body) : "{}";
 
-    return validateContactUsFormBody(bodyAsJson)
+    const passedCaptchaValidation = await validateCaptchaToken(
+      `${bodyAsJson.captchaToken}`
+    );
+    return validateContactUsFormBody(bodyAsJson) && passedCaptchaValidation
       ? buildContactUsReqBody(bodyAsJson)
       : undefined;
   } catch (error) {
     return undefined;
   }
 };
 
+const validateCaptchaToken = async (token: string) => {
+  const recaptchaSecret = (await recaptchaConfigPromise)?.secretKey;
```

Comment:
Can we fail the check before making the validation fetch if `recaptchaSecret` is undefined?